p2p/dandelion-tower/src/pool/incoming_tx.rs

```diff
@@ -0,0 +1,110 @@
+//! Contains [`IncomingTx`] and [`IncomingTxBuilder`]
+use crate::{State, TxState};
+
+/// An incoming transaction that has gone through the preprocessing stage.
+pub struct IncomingTx<Tx, TxID, PID> {
+    /// The transaction.
+    pub(crate) tx: Tx,
+    /// The transaction ID.
+    pub(crate) tx_id: TxID,
+    /// The routing state of the transaction.
+    pub(crate) routing_state: TxState<PID>,
+}
+
+/// An [`IncomingTx`] builder.
+///
+// The const generics here are used to restrict what methods can be called.
+pub struct IncomingTxBuilder<const RS: bool, const DBS: bool, Tx, TxID, PID> {
```

Comment:
Can you document what `RS` and `DBS` are or change them to more obvious names like `ROUTING_STATE` and `DATABASE_STATE` or `DB_STATE`? I'm not sure what `PID` is either, pool ID?

If these are used elsewhere maybe add a section in the `lib.rs` docs instead.

Reply:
I added docs above, these are just an implementation detail so probably don't need to be in public docs